### PR TITLE
Filter the placeholders to show them with the glossary words

### LINKED ISSUE #1400: Edge case? Placeholders & glossary words.
https://translate.wordpress.org/projects/wp-plugins/wordpress-seo/dev/nl-be/default/?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=9836002&filters%5Btranslation_id%5D=93348563

In nl_BE, we have an entry for "Show" in our glossary. In the above string, we have a placeholder that ends on "s" and the rest of the string continues with "how"...

Somehow, GlotPress sees this as a glossary word while it shouldn't.

### PR DESCRIPTION
**Pull request by amieiro**
## Problem

Imagine we have the `show` word in the glossary. If we have a placeholder in an original string finishing with the `s` letter and, without space, we have the `how` word: `%show`, the current algorithm detects the glossary word and not the placeholder, showing on mouse over a popover with the glossary information, instead of detects the `%s` placeholder and the `how` word.

- Current approach: `%show` -> `%` + `show`  
- Proposed approach: `%show` -> `%s` + `how`

Fixes https://github.com/GlotPress/GlotPress/issues/1400.

## Solution

This solution updates the regex that split the original string into an array, to isolate the placeholders, avoiding mixing parts of them with other substrings. 

The regex used is the same used to [detect warnings](https://github.com/GlotPress/GlotPress/blob/06a146e42fff13cea2233ef025f9d0fa4370e208/gp-includes/warnings.php#L377) with the placeholders.

## Testing Instructions

This PR adds two tests in the `tests/phpunit/testcases/test_template_helper_functions.php` file:

- `test_map_glossary_entries_with_placeholders_glued_glossary_words`
- `test_map_glossary_entries_with_placeholders_glued_glossary_words_in_the_plural_origin`

### CODE REVIEW

File: gp-templates/helper-functions.php [hunk 1]
```diff
@@ -213,20 +213,18 @@ function map_glossary_entries_to_translation_originals( $translation, $glossary
 		}
 
 		// Build the regular expression.
-		$terms_search = '\b(';
+		$terms_search = '((?<!%)%(?:[1-9]\$)?[bcdefgosuxEFGX%l@])|(\b';
```
Comment (akirk) [inline gp-templates/helper-functions.php]: For me the test also passes like this, can we remove it?
```suggestion
		$terms_search = '(%(?:[1-9]\$)?[bcdefgosuxEFGX%l@])|(\b';
```

Comment (amieiro) [reply gp-templates/helper-functions.php]: I have added a new check in the current tests to address this question and this part is not needed, so I have committed this improvement.

**Comment by amieiro** [comment]
To test the overhead in load time that this new regex might add, I used a Bash script to test it:

Before, I have: 
- Increased the number of items in an anonymous page from 15 to 500, changing this value [here](https://github.com/GlotPress/GlotPress/blob/7e98b14ad5fde547c89ca50f4261d0ab717d5912/gp-includes/things/translation.php#L22).
- Added a [real glossary](https://translate.wordpress.org/locale/gl/default/glossary/) with 293 items.

This is the script (change your local URL to test it):

```
#!/bin/bash

# Define the URL of the local web page
URL="https://glotpress.test/glotpress/projects/local-wp/dev/gl/default/"

# Number of times to measure the load time
NUM_ITERATIONS=100

# Initialize a variable to store the total load time
TOTAL_TIME=0

# Loop through the specified number of iterations
for ((i=1; i<=$NUM_ITERATIONS; i++)); do
    # Use curl to fetch the URL and measure the time it took
    load_time=$(curl -s -o /dev/null -w "%{time_total}\n" "$URL")
    
    # Add the load time to the total
    TOTAL_TIME=$(echo "$TOTAL_TIME + $load_time" | bc)
    
    # Print the load time for this iteration
    echo "Load time for iteration $i: ${load_time} seconds"
done

# Calculate the mean load time in milliseconds
mean_time=$(echo "scale=3; ($TOTAL_TIME / $NUM_ITERATIONS) * 1000" | bc)

# Print the mean load time in milliseconds
echo "Mean load time: ${mean_time} milliseconds"
```
This is the output after 100 executions:
- Before: Mean load time: 781.000 milliseconds
- With this PR: Mean load time: 778.000 milliseconds

I have obtained similar outputs with another executions, so no overload is apparent.

**Review by akirk — APPROVED** [review]
Thank you for doing the performance tests, looks good if it performs at the same speed!

**Pull request merged**